controllers/openstackcluster_controller.go

```diff
@@ -158,7 +160,8 @@ func (r *OpenStackClusterReconciler) reconcileDelete(ctx context.Context, log lo
 		log.Info("OpenStack router deleted successfully")
 	}
 
-	if openStackCluster.Status.Network != nil {
+	// if NodeCIDR was not set, no network was created.
+	if openStackCluster.Status.Network != nil && openStackCluster.Spec.NodeCIDR != "" {
```

Comment:
Just that I get it correctly. We were using an already existing network and that's why we don't want to delete it via CAPO?

Reply:
Yes that is correct. In my setup I use an existing network that is shared with other VMs. I think it make sense to not clean resources that are not allocated by the controller.

Reply:
Yup makes sense